### PR TITLE
refactor: combine CompileClient and CompileClientActor

### PR DESCRIPTION
**Pull request by QuarticCat**

### CODE REVIEW

File: crates/tinymist/src/actor/export.rs [hunk 1]
```diff
@@ -78,7 +80,12 @@ impl ExportActor {
             'accumulate: loop {
                 log::debug!("RenderActor: received request: {req:?}");
                 match req {
-                    ExportRequest::ChangeConfig(cfg) => self.config = cfg,
+                    ExportRequest::ChangeConfig(config) => {
```
Comment (Myriad-Dreamin) [inline crates/tinymist/src/actor/export.rs:83]: I think we can remove the entry field from `ExportConfig` for more clear code, if whatever we start to tend to update config separately from entry here.

File: crates/tinymist/src/actor/typ_server.rs [hunk 2]
```diff
@@ -99,17 +79,17 @@ pub struct CompileServerActor<C: Compiler> {
...
     latest_success_doc: Option<Arc<TypstDocument>>,
     /// feature set for compile_once mode.
     once_feature_set: Arc<FeatureSet>,
     /// Shared feature set for watch mode.
     watch_feature_set: Arc<FeatureSet>,
 
-    /// Internal channel for stealing the compiler thread.
-    steal_tx: mpsc::UnboundedSender<Interrupt<Self>>,
-    steal_rx: mpsc::UnboundedReceiver<Interrupt<Self>>,
+    /// Channels for stealing the compiler thread.
```
Comment (Myriad-Dreamin) [inline crates/tinymist/src/actor/typ_server.rs:90]: The stealing interrupt only refers to the `Interrupt::Task`.
We can update the document to:
```rs
/// Channels for sending interrupts to the compiler thread.
```

File: crates/tinymist/src/actor/typ_client.rs [hunk 3]
```diff
@@ -286,42 +284,81 @@ impl CompileClientActor {
         diag_group: String,
         config: CompileConfig,
         entry: EntryState,
-        inner: Deferred<CompileClient>,
+        steal_tx: mpsc::UnboundedSender<Interrupt<CompileService>>,
```
Comment (Myriad-Dreamin) [inline crates/tinymist/src/actor/typ_client.rs:287]: rename `steal_tx` to `intr_tx`

**Comment by QuarticCat** [comment]
All resolved.

**Pull request merged**